### DISCUSSION TITLE
Cancel Request Handler on Client Disconnect

### DISCUSSION THREAD

**Discussion by winstxnhdw** [emmett-framework/granian#417] [Ideas]
I am wondering if handling `http.disconnect` is something that Granian will want to support or should it be implemented at a higher level?

**Comment by gi0baro** [comment]
@winstxnhdw can you elaborate on what you mean by *higher level*?

**Reply by winstxnhdw** [reply]
Should the responsibility for handling this behaviour be at whatever framework or application that adopts Granian.

**Comment by gi0baro** [comment]
To clear this out: Granian currently supports and emits `http.disconnect` events, as part of the ASGI spec.

The main point of discussion (in #286) regards the fact Granian is currently unable to emit such event eagerly when the client actually disconnects. And as stated in #286, this is not really a matter of *willingness* in supporting such behaviour, but rather a consequence of the structure in the underlying library which Granian uses.
So, to answer your question, the only ways to *fix* #286 are either:
- have a proper indication from Hyper team on how to do this with 1.x (if possible at all)
- have Hyper team add the functionality in the library and thus implement the relevant code in Granian

Given that #286 is open for pledges, I will be happy to accepts PRs implementing that and give 100% of the pledges to the relevant author(s).

I don't see any other way to proceed here.
